marquez_client/client.py

```diff
@@ -9,243 +9,261 @@
 
     @property
-    def namespace(self):
+    def namespace_name(self):
         return self._namespace_name
 
-    #  Namespaces API
-    @handle_response
+    def create_namespace(self, namespace_name, owner_name, description=None):
+        if not namespace_name:
+            raise ValueError('namespace_name must not be None')
+        if not owner_name:
+            raise ValueError('owner_name must not be None')
+
+        return self._put(
+            self._url('/namespaces/{0}', namespace_name),
+            payload={
+                'owner': owner_name,
+                'description': description
+            })
+
     def get_namespace(self, namespace_name):
         if not namespace_name:
-            raise ValueError("Please provide a namespace")
-        path = "namespaces/{0}"
-        path_args = [namespace_name]
-        return self.get_request(path, path_args)
-
-    @handle_response
-    def list_namespaces(self):
-        path = "namespaces"
-        path_args = []
-        return self.get_request(path, path_args)
-
-    @handle_response
-    def create_namespace(self, name, owner,
-                         description=None):
-        path = "namespaces/{0}"
-        path_args = [name]
-        payload = {
-            "owner": owner,
-        }
-        if description:
-            payload['description'] = description
-        return self.put_request(path, path_args, payload)
-
-    #  Datasources API
-    @handle_response
-    def list_datasources(self, limit=None, offset=None):
-        path = "datasources"
-        path_args = []
-        request_args = {}
-        if limit:
-            request_args['limit'] = limit
-        if offset:
-            request_args['offset'] = offset
-        return self.get_request(
-            path, path_args, request_args)
-
-    @handle_response
-    def get_datasource(self, urn):
-        path = "datasources/{0}"
-        path_args = [urn]
-        return self.get_request(path, path_args)
-
-    @handle_response
-    def create_datasource(self, name, connection_url):
-        path = "datasources"
-        path_args = []
-        payload = {
-            "name": name,
-            "connectionUrl": connection_url
-        }
-        return self.post_request(path, path_args, payload)
-
-    #  Jobs API
-    @handle_response
-    def create_job(self, job_name, location, input_dataset_urns,
-                   output_dataset_urns, namespace_name=None,
-                   description=None):
+            raise ValueError('namespace_name must not be None')
+
+        return self._get(self._url('/namespaces/{0}', namespace_name))
+
+    def list_namespaces(self, limit=None, offset=None):
+        return self._get(
+            self._url('/namespaces'),
+            params={
+                'limit': limit,
+                'offset': offset
+            })
+
+    def create_job(self, job_name, location, input_dataset_urns=[], output_dataset_urns=[], description=None, namespace_name=None):
```

Comment:
A word of caution about initializing `input_dataset_urns` and `output_dataset_urns` as `[]`. What happens is that these values persist between invocations and changes to these lists will be reflected in future calls.
https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html

It's recommended to use `None` instead.

Reply:
Fixed, and also now aware of this anti-pattern 👍